### PR TITLE
Remove crypt support from ansible.utils.encrypt

### LINKED ISSUE #81717: Remove crypt support from ansible.utils.encrypt for 2.17
### Summary

The `CryptHash` class is scheduled for removal for 2.17

### Issue Type

Feature Idea

### Component Name

`lib/ansible/utils/encrypt.py`

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY

Fixes #81717

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

**Review by bcoca — APPROVED** [review]

File: test/integration/targets/become/aliases [hunk 1]
```diff
@@ -2,3 +2,4 @@ destructive
 shippable/posix/group1
 context/target
 gather_facts/no
+setup/always/setup_passlib_controller
```
Comment (mkrizek) [inline test/integration/targets/become/aliases]: @mattclay I added this for every target that uses `setup_test_user`. I feel like `setup_passlib_controller` should be run as a dependency/part of `setup_test_user` but I couldn't make it work for `setup_passlib_controller` to be run always on a controller side as `setup_test_user` is used either as a meta dependency of `targets` or as `needs/target/setup_test_user` and called explicitly in a test either in `roles:` or via `include_role`.

Comment (mattclay) [reply test/integration/targets/become/aliases]: Yeah, currently `setup/always/` doesn't work recursively. It's detected as a dependency, but it's not actually invoked.

Comment (mkrizek) [reply test/integration/targets/become/aliases]: I added a note that it's required by `setup_test_user`, at least.

File: test/integration/targets/setup_passlib_controller/runme.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+set -eux
+
+# Requirements have to be installed prior to running ansible-playbook
+# because plugins and requirements are loaded before the task runs
+python -m pip install passlib
```
Comment (mattclay) [inline test/integration/targets/setup_passlib_controller/runme.sh:12]: This isn't using a virtual environment, which is going to be problematic in the not-too-distant future. It's probably time for me to add support for ansible-test managed virtual environments for integration tests, which would greatly simplify this and avoid the issue of working without a virtual environment.

File: test/integration/targets/setup_passlib_controller/runme.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+set -eux
+
+# Requirements have to be installed prior to running ansible-playbook
```
Comment (mattclay) [inline test/integration/targets/setup_passlib_controller/runme.sh:10]:
```suggestion
# Temporary hack for PEP 668 on newer systems.
# Remove once ansible-test can provide targets their own virtual environment.
# Tests which can manage their own virtual environment should not use this approach.
export PIP_BREAK_SYSTEM_PACKAGES=1

# Requirements have to be installed prior to running ansible-playbook
```

**Review by mattclay — CHANGES_REQUESTED** [review]

**Review by mattclay — APPROVED** [review]

File: test/units/utils/test_encrypt.py [hunk 4]
```diff
@@ -35,31 +21,12 @@ def assert_hash(expected, secret, algorithm, **settings):
         assert excinfo.value.args[0] == "passlib must be installed and usable to hash with '%s'" % algorithm
```
Comment (webknjaz) [inline test/units/utils/test_encrypt.py:21]: @mkrizek the entire else-branch here lost coverage with the changes in this PR. Should the assertion here be made unconditional or new tests added?

Comment (webknjaz) [reply test/units/utils/test_encrypt.py:21]: See https://app.codecov.io/gh/ansible/ansible/commit/6935c8e303440addd3871ecf8e04bde61080b032/indirect-changes#3ab5fa106f6a24247d92fd39c8e79a4f-R19

Comment (webknjaz) [reply test/units/utils/test_encrypt.py:21]: And this is the only place with dead code in this test module: https://app.codecov.io/gh/ansible/ansible/blob/devel/test%2Funits%2Futils%2Ftest_encrypt.py

**Pull request merged**